python/brayns/core/geometry/bounded_plane.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) 2015-2022 EPFL/Blue Brain Project
+# All rights reserved. Do not distribute without permission.
+#
+# Responsible Author: nadir.romanguerrero@epfl.ch
+#
+# This file is part of Brayns <https://github.com/BlueBrain/Brayns>
+#
+# This library is free software; you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License version 3.0 as published
+# by the Free Software Foundation.
+#
+# This library is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this library; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from multiprocessing.sharedctypes import Value
+from typing import Any
+
+from .geometry import Geometry
+from .box import Box
+
+
+@dataclass
+class BoundedPlane(Geometry):
+    """Axis-aligned bounded plane.
+
+    Described by the equation ``ax + by + cz + d = 0``.
+
+    Where [a, b, c] is the normal of the plane and d the orthogonal distance
+    from the origin.
+
+    :param a: X term of the plane equation.
+    :type a: float
+    :param b: Y term of the plane equation.
+    :type b: float
+    :param c: Y term of the plane equation.
+    :type c: float
+    :param d: Scalar term of the plane equation.
+    :type d: float
+    :param bounds: Axis aligned bounds to limit the plane.
+    :type bounds: Box
+    """
+
+    a: float
+    b: float
+    c: float
+    d: float
+    bounds: Box
+
+    def __new__(cls, a: float, b: float, c: float, d: float, bounds: Box) -> BoundedPlane:
```

Comment:
You can override __post_init__ like this:

```
def __post_init__(self) -> None:
    if ... raise ValueError()
```

Overriding new is only for very specific cases (mostly when inheriting from built-in types.

The advantage of __post_init__ is that you don't need to duplicate the a, b, c, d, bounds and to call the __new__.

Reply:
__new__ allows me to throw the exception before the object is actually built, thats why I used it.